jdaviz/core/tools.py

```diff
@@ -144,8 +144,24 @@ def on_update_zoom(self):
             # let's ignore and reset the tool
             return
 
-        self.viewer.state.x_min, self.viewer.state.x_max = self.interact.selected_x
+        new_interact_x = self.interact.selected_x
+
+        # If aspect is equal, then we need to preserve that aspect ratio
+        # even if the box zoom area does not.
+        if self.viewer.state.aspect == "equal":
+            aspect_ratio = ((self.viewer.state.x_max - self.viewer.state.x_min) /
+                            (self.viewer.state.y_max - self.viewer.state.y_min))
+
+            x_diff = self.interact.selected_x[1] - self.interact.selected_x[0]
+            y_diff = self.interact.selected_y[1] - self.interact.selected_y[0]
```

Comment:
can `[1]` ever be smaller than `[0]` (inverted axes), and would that cause problems?  I suspect we wouldn't see that in jdaviz for any case that uses equal aspect ratio, so maybe just an inline comment stating if this assumption is made would suffice for now.

Reply:
I tried to box zoom backwards and `[1]` is always the max value. I will add a comment to clarify that.